File: EXP-Rewrite/src/main/java/ca/exp/soundboard/rewrite/soundboard/KeyEventIntConverter.java
package ca.exp.soundboard.rewrite.soundboard;

import java.awt.Toolkit;
import java.awt.event.KeyEvent;

public class KeyEventIntConverter {
    public static String getKeyEventText(int keyCode) {
        if (keyCode >= 96 && keyCode <= 105) {
            String numpad = Toolkit.getProperty("AWT.numpad", "NumPad");
            char character = (char) (keyCode - 96 + 48);
            return numpad + " " + character;
        }

        return KeyEvent.getKeyText(keyCode);
    }
}

File: src/main/java/ca/exp/soundboard/soundboard/KeyEventIntConverter.kt
package ca.exp.soundboard.soundboard

import java.awt.Toolkit
import java.awt.event.KeyEvent

object KeyEventIntConverter {

    @JvmStatic
    fun getKeyEventText(keyCode: Int): String {
        if (keyCode in 96..105) {
            val numpad = Toolkit.getProperty("AWT.numpad", "NumPad")

            val character = (keyCode - 96 + 48).toChar()
            return "$numpad $character"
        }

        return KeyEvent.getKeyText(
            keyCode
        )
    }
}
